src/lustre/lustreDesugarFrameBlocks.ml

```diff
@@ -0,0 +1,264 @@
+  | NArityOp (a, b, l) -> NArityOp (a, b, List.map (fill_ite_helper fill) l) 
+  | Call (a, b, l) -> Call (a, b, List.map (fill_ite_helper fill) l)
+  | CallParam (a, b, c, l) -> CallParam (a, b, c, List.map (fill_ite_helper fill) l)
+
+  | Merge (a, b, l) -> Merge (a, b, 
+    List.combine
+    (List.map fst l)
+    (List.map (fill_ite_helper fill) (List.map snd l)))
+  
+  | RecordExpr (a, b, l) -> RecordExpr (a, b,     
+    List.combine
+    (List.map fst l)
+    (List.map (fill_ite_helper fill) (List.map snd l)))
+  
+  | RestartEvery (a, b, l, e) -> 
+    RestartEvery (a, b, List.map (fill_ite_helper fill) l, fill_ite_helper fill e)
+  | Activate (a, b, e, r, l) ->
+    Activate (a, b, (fill_ite_helper fill) e, (fill_ite_helper fill) r, List.map (fill_ite_helper fill) l)
+  | Condact (a, e, r, b, l1, l2) ->
+    Condact (a, (fill_ite_helper fill) e, (fill_ite_helper fill) r, b, 
+             List.map (fill_ite_helper fill) l1, List.map (fill_ite_helper fill) l2)
+
+  | StructUpdate (a, e1, li, e2) -> 
+    A.StructUpdate (a, fill_ite_helper fill e1, 
+    List.map (function
+              | A.Label (a, b) -> A.Label (a, b)
+              | Index (a, e) -> Index (a, fill_ite_helper fill e)
+             ) li, 
+    fill_ite_helper fill e2)
+
+(** Helper function to generate node equations when an initialized variable in the 
+    frame block is left undefined in the frame block body. *)
+let generate_undefined_nes nis ne = match ne with
+  | A.Equation (pos, (StructDef(_, [SingleIdent(_, id)]) as lhs), init) -> 
+    (* Find the corresponding node item in frame block body. *)
+    let res = List.find_opt (fun ni -> match ni with
+      | A.Body (A.Equation (_, StructDef(_, [SingleIdent(_, i)]), _)) when id = i -> true
+      | _ -> false
+    ) nis in (
+    match res with
+      (* Already defined in frame block *)
+      | Some _ -> R.ok []
+      (* Fill in equation in frame block body *)
+      | None -> R.ok [A.Body(A.Equation(pos, lhs, Arrow(pos, init, Pre(pos, Ident (pos, id)))))]
+    )
+  | A.Equation (pos, (StructDef(_, [ArrayDef(_, id1, id2)]) as lhs), init) -> 
+    (* Find the corresponding node item in frame block body. *)
+    let res = List.find_opt (fun ni -> match ni with
+      | A.Body (A.Equation (_, StructDef(_, [ArrayDef(_, i, _)]), _)) when id1 = i -> true
+      | _ -> false
+    ) nis in 
+    let rec build_array_index js = (match js with
+      | [j] -> A.ArrayIndex(pos, A.Ident(pos, id1), A.Ident(pos, j))
+      | j :: js -> ArrayIndex(pos, build_array_index js, A.Ident(pos, j))
+      | [] -> assert false (* not possible *)
+    ) in
+
+    (match res with
+      (* Already defined in frame block *)
+      | Some _ -> R.ok []
+      (* Fill in equation in frame block body *)
+      | None -> R.ok [A.Body(A.Equation(pos, lhs, Arrow(pos, init, Pre(pos, build_array_index (List.rev id2)))))]
+    )
+  (* Assert in frame block guard *)
+  | A.Equation(pos, _, _) -> mk_error pos (MisplacedNodeItemError (A.Body ne))
```

Comment:
My understanding is that this case is impossible because multiple assignments in frame blocks have been desugared. If that is the case, I would replace the call to `mk_error` with `assert false` and I would explain why the case is impossible with a comment.